PROJ-115 :: Added retro for sprint 3 (#63)
SW805F20/report: sections/sprint-3/conclusion.tex

In the future, a single person should start with the development and when someone is available to join at a later point, they can.

\subsubsection*{How is the daily stand-up working?}
While the new format proposed in the last sprint's retrospective was effective for a few days, it was quickly forgotten.
To keep the stand-ups effective, the anchor should be better at enforcing that the format is being followed to make sure that we know who needs help, and what people are currently doing.

\subsubsection*{How are the reviews going?}
The pair review format for code does not seem to be working too well since it is usually procrastinated until the next morning because one of the two reviewers is busy with something else.

The single-person reviews are also lacking at the moment, where people do not remember to check what they are assigned to.
It is suggested that people regularly go to the front page of GitHub and check the "Recent Activity" which shows the PRs you are assigned to.
It is also suggested that it is possible to ask people to review your code or text right away if it is important that it gets merged.

> This is mostly if the task is blocking other tasks and thus are of great importance.

Reviewer: ```suggestion
This is mostly if the task is blocking other tasks and thus is of great importance.
```

\section{Sprint 3 conclusion}\label{sec:sprint3conclusion}
This section concludes the preceding chapter on sprint 3.

> It will conclude the knowledge that was obtained based on the sections of the chapter, and discuss the retrospective meeting that was held in the end of the sprint.

Reviewer 2: ```suggestion
It will recapitulate the knowledge that was obtained based on the sections of the chapter, and discuss the retrospective meeting that was held in the end of the sprint.
```

This overview will be split into two categories, to reflect the structure of the project: Networking and game.

\subsubsection{Networking}
The major focus for the network part this sprint was to generate an initial UPPAAL model to visualize the flow of data.
This initial model is based on the assumption that packages are not lost and timeouts will not happen for simplicity.
Creation of this model lead to some revision to the network protocol as described in \autoref{subsec:sprint3networkupdate}.\\
In addition to the UPPAAL model, the ideas behind dead reckoning was examined in relation to this project.
The technique was deemed relevant to implement if the representation of the players' movement is inconsistent, which will be delved further into when the game is connected with the actual Pozyx location data.

\subsubsection{Game}
The most major changes in terms of the game this sprint was the addition of goal zones.
The goal zones were previously generated on the client, but to ensure a consistent game state across all clients, this has been moved to the host and will be transmitted over the network instead.

> This meant that the major focus for the game in this sprint was refactoring how goals were displayed, as described in \ref{subsec:goalrefactoring}.

Reviewer 2: ```suggestion
This meant that the major focus for the game in this sprint was refactoring how goals were displayed, as described in \autoref{subsec:goalrefactoring}.
```

\begin{figure}[H]
	\centering
	\includegraphics[width=0.6\linewidth]{sprint3/gamestatus.png}
	\caption{The current game status with goals and textures}
	\label{fig:sprint3status}
\end{figure}
\noindent
Additionally, some temporary textures have been added to the game to get rid of the pink playing field and square players.
These new textures can be seen on \autoref{fig:sprint3status}.

\subsection{Retrospective on the process}

> The hoped outcome of this sprint was an MVP of the product, but unfortunately, this was not achieved due to various problems.

Reviewer 2: ```suggestion
The ideal outcome of this sprint was an MVP of the product, but unfortunately, this was not achieved due to various problems.
```

First of all, there was a series of problems with merge conflicts in auto-generated files in Unity, which lead to the agreement that only one pair of group members should work on Unity at a time to avoid unsolvable merge conflicts.
This lead to a major delay in the Unity development.\\

> Additionally, due to the COVID-19 lockdown, we had not had an opportunity to test the system properly with external users to get their feedback on the current product, and what to focus on for the last two sprints.

Reviewer 2: ```suggestion
Additionally, due to the COVID-19 lockdown, we did not have an opportunity to test the system properly with external users to get their feedback on the current product, and what to focus on for the last two sprints.
```

Reviewer 2: Det virker lidt som om at det kommer ud af det blå. Har vi tidligere skrevet at det var et mål at teste systemet i dette sprint?

Når jeg læser det kunne man godt forstå det som, at hvis COVID-19 ikke var der, og at vi havde mulighed for at teste produktet, så kunne vi have testet systemet i denne periode.

> 

Reviewer 2: ```suggestion
```

In addition to discussing the progress of the process, the following points were brought up:

\subsubsection*{How does the process with Jira work?}
The process is generally improved since last time, not having to spend time assigning priority points to every task is nice.

> The challenger expressed a wish that he would appreciate it if more people contributed to the backlog suggestions since it can be difficult to continuously come up with new tasks of things that need to be done.

Reviewer 2: ```suggestion
The challenger expressed that he would appreciate if more people contributed to the backlog suggestions since it can be difficult to continuously come up with new tasks.
```

\subsubsection*{How is the daily stand-up working?}
While the new format proposed in the last sprint's retrospective was effective for a few days, it was quickly forgotten.
To keep the stand-ups effective, the anchor should be better at enforcing that the format is being followed to make sure that we know who needs help, and what people are currently doing.

\subsubsection*{How are the reviews going?}
The pair review format for code does not seem to be working too well since it is usually procrastinated until the next morning because one of the two reviewers is busy with something else.

The single-person reviews are also lacking at the moment, where people do not remember to check what they are assigned to.
It is suggested that people regularly go to the front page of GitHub and check the "Recent Activity" which shows the PRs you are assigned to.
It is also suggested that it is possible to ask people to review your code or text right away if it is important that it gets merged.
This is mostly if the task is blocking other tasks and thus are of great importance.

> \subsubsection*{Other comments?}

Reviewer 2: Synes måske ikke at denne sektion er nødvendig.

Author: nå
